functions/go/search-replace/README.md

```diff
@@ -0,0 +1,179 @@
+# search-replace
+
+## Overview
+
+<!--mdtogo:Short-->
+
+Search and optionally replace fields across all resources.
+
+<!--mdtogo-->
+
+There is a spectrum of configuration customization techniques as described in
+[this document]. One of the most basic and simplest to understand is
+Search and Replace: The user fetches a package of configuration, searches all
+the files for fields matching a criteria, and replaces their values.
+
+Search matchers are provided with `by-` prefix. When multiple matchers
+are provided they are AND’ed together. `put-` matchers are mutually exclusive.
+
+### Synopsis
+
+<!--mdtogo:Long-->
+
+```
+kpt fn eval gcr.io/kpt-fns/search-replace:VERSION [DIR] -- [matcher_name=matcher_value]
+```
+
+#### Matchers
+
+```sh
+by-value
```

Comment:
What's this shell block supposed to be? Do you want a list instead?

Reply:
`sh` is not needed. Removed.